src/Pages/Tool.cshtml.cs

```diff
@@ -208,20 +212,40 @@ public class ToolModel : PageModel
 
             if (messageType == Constants.Lti.LtiDeepLinkingRequestMessageType)
             {
-                return Post("./Catalog", new { idToken });
+                return Post("/Catalog", new { idToken });
             }
 
             IdToken = idToken;
             LtiRequest = new LtiResourceLinkRequest(jwt.Payload);
 
+            var tokenResponse = await _accessTokenService.GetAccessTokenAsync(
+                LtiRequest.Iss, 
+                Constants.LtiScopes.Ags.LineItem);
+
+            var lineItemClient = _httpClientFactory.CreateClient();
+            lineItemClient.SetBearerToken(tokenResponse.AccessToken);
+            lineItemClient.DefaultRequestHeaders.Accept
+                .Add(new MediaTypeWithQualityHeaderValue(Constants.MediaTypes.LineItem));
+
+            var resultsUrl = $"{LtiRequest.AssignmentGradeServices.LineItemUrl}/{Constants.ServiceEndpoints.Ags.ResultsService}";
```

Comment:
this breaks if the platform includes query params in the uri, should use `var resultsUrl = $"{lineItemUri.GetLeftPart(UriPartial.Path)}/Constants.ServiceEndpoints.Ags.ResultsService{lineItemUri.Query}";` instead